examples/nlp/mlm_and_finetune_with_bert.py

```diff
@@ -0,0 +1,290 @@
+"""
+Title: End to End Masked Language Modeling & Fine-Tuning with BERT from Scratch
+Author: [Ankur Singh](https://twitter.com/ankur310794)
+Date created: 2020/09/03
+Last modified: 2020/16/03
+Description: Implement a Masked Language Modeling with BERT and fine-tune on IMDB Reviews dataset
+"""
+"""
+## Introduction
+Masked language modeling is a fill-in-the-blank task, where a model uses the context words surrounding a [MASK] token to try to predict what the [MASK] word should be.
+"""
+
+"""
+## Setup
+Install tf-nightly via pip install tf-nightly.
+"""
+
+import tensorflow as tf
+from dataclasses import dataclass
+import pandas as pd
+from tensorflow.keras.layers.experimental.preprocessing import TextVectorization
+import numpy as np
+
+"""
+## Set-up Configuration
+"""
+
+
+@dataclass
+class Config:
+    MAX_LEN = 128
+    BATCH_SIZE = 16
+    LR = 1e-5
+    VOCAB_SIZE = 20000
+    EMBED_DIM = 32
+    NUM_HEAD = 2  # used in bert model
+    FF_DIM = 32  # used in bert model
+
+
+flags = Config()
+
+"""
+## Load Data
+"""
+
+"""shell
+wget https://raw.githubusercontent.com/SrinidhiRaghavan/AI-Sentiment-Analysis-on-IMDB-Dataset/master/imdb_tr.csv
+"""
+
+data = pd.read_csv("imdb_tr.csv", encoding="ISO-8859-1")
+
+"""
+## Dataset Preparation
+"""
+
+
+def get_vectorize_layer(texts, vocab_size, max_seq, special_tokens=["[MASK]"]):
+    """Build Text vectorization layer
+
+    Args:
+        texts (list): List of String i.e input texts
+        vocab_size (int): vocab size
+        max_seq (int): maximum sequence len
+        special_tokens (list, optional): List of special tokens. Defaults to ['[MASK]'].
+
+    Returns:
+        tf.keras.layers.Layer: Return TextVectorization Keras Layer
+    """
+    vectorize_layer = TextVectorization(
+        max_tokens=vocab_size,
+        output_mode="int",
+        standardize=None,
+        output_sequence_length=max_seq,
+    )
+
+    vectorize_layer.adapt(texts)
+
+    vocab = vectorize_layer.get_vocabulary()
+    vocab = vocab[2 : vocab_size - len(special_tokens)] + ["[MASK]"]
```

Comment:
If you put it as the start of the list you won't need to compute `mask_token_id` by hand (it will be 0)

Reply:
I am a little bit confused. My understanding is index 0 is reserved for padding token. ([Reference](https://github.com/tensorflow/tensorflow/blob/458bb34ae6059440af847e1c10bd7db78ae568ea/tensorflow/python/keras/layers/preprocessing/text_vectorization.py#L272))

And I feel there is no argument called mask_token in the class constructor of TextVectorization ([Reference](https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/keras/layers/preprocessing/text_vectorization.py#L295))

I need to insert special tokens ('[MASK] is this case') into the vocabulary and I would like to use zero index for padding. 

Am I thinking right?